app/views.py

```diff
@@ -0,0 +1,339 @@
+from app import app, db
+from app.models import User, BucketList, ListItem
+from flask_jwt import JWT, jwt_required, current_identity
+from werkzeug.exceptions import HTTPException, default_exceptions
+from flask import request, g, jsonify, abort
+from app.validate import validate_register, validate_bucketlist, validate_item\
+    , validate_limit_and_offset
+
+
+def verify_password(email, password):
+    g.user = User.query.filter(email == User.email).first()
+    if g.user is not None and g.user.check_password(password):
+        return g.user
+    return False
+
+
+def identity(payload):
+    user_id = payload['identity']
+    return User.query.filter(user_id == User.user_id).first()
+
+
+jwt = JWT(app, verify_password, identity)
+
+
+def user_exists(email):
+    exists = db.session.query(db.session.query(User)
+                              .filter_by(email=email)
+                              .exists()).scalar()
+    return exists
+
+
+@app.route("/resource", methods=['POST'])
+@jwt_required()
+def get_resource():
+    return jsonify(dict(msg="Hello world"))
+
+
+def handle_error(error):
+    response = {}
+    status_code = 500
+    if isinstance(error, HTTPException):
+        status_code = error.code
+    response["status_code"] = status_code
+    response["error"] = str(error)
+    response['description'] = error.description
+    return jsonify(response), status_code
+
+
+# change error handler to return json instead of html
+for code in default_exceptions.keys():
+    app.errorhandler(code)(handle_error)
+
+
+@app.route("/auth/register", methods=['POST'])
+def register_user():
+    response = {}
+    json = request.json
+    validation = validate_register(json)
+    if validation.status:
+        if not user_exists(json["email"]):
+            user = User(
+                first_name=json["first_name"],
+                last_name=json["last_name"],
+                email=json["email"],
+                password=json["password"]
+            )
+            db.session.add(user)
+            db.session.commit()
+            status_code = 201
+            response["message"] = \
+                "Successfully registered the user with the email %s." % (json[
+                                                                         "email"])
+        else:
+            status_code = 409
```

Comment:
Check what status 409 mean from the doc above and you will find that it might not be the most appropriate.